pandas/tests/groupby/transform/test_numba.py

```diff
@@ -181,10 +181,24 @@ def f(values, index):
 
     df = DataFrame({"group": ["A", "A", "B"], "v": [4, 5, 6]}, index=[-1, -2, -3])
     result = df.groupby("group").transform(f, engine="numba")
-    expected = DataFrame([-4.0, -3.0, -2.0], columns=["v"], index=[-1, -2, -3])
+    expected = DataFrame([-2.0, -3.0, -4.0], columns=["v"], index=[-1, -2, -3])
     tm.assert_frame_equal(result, expected)
 
 
+def test_index_order_consistency_preserved():
+    # GH 57069
+    pytest.importorskip("numba")
+
+    def f(values, index):
+        return values
+
+    df = DataFrame({"vals": [0.0, 1.0, 2.0, 3.0], "group": [0, 1, 0, 1]})
+    df.index = df.index.values[::-1]
+    result = df.groupby("group")["vals"].transform(f, engine="numba")
+    expected = Series([0.0, 1.0, 2.0, 3.0], index=[3, 2, 1, 0], name="vals")
```

Comment:
```suggestion
    expected = Series([0.0, 1.0, 2.0, 3.0], index=range(3, -1, -1), name="vals")
```

Reply:
Thank you for the reviews, I've applied the suggestions.